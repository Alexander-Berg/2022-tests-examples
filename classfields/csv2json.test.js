const csv2json = require('./csv2json');

const fs = require('fs');
const path = require('path');

const questionsCsv = fs.readFileSync(path.resolve(__dirname, '../mocks/questions.csv'), 'utf8');

it('должен сформировать объект c вопросами из csv файла', () => {
    expect(csv2json(questionsCsv)).toMatchSnapshot();
});
