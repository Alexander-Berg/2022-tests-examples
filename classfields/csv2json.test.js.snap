// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен сформировать объект c вопросами из csv файла 1`] = `
Object {
  "paramColumns": Array [
    "body_type_group",
    "search_tag",
  ],
  "questions": Array [
    Object {
      "answers": Array [
        Object {
          "answer": "От",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "price_from",
          "search_tag": Array [
            "",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
        Object {
          "answer": "До",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "price_to",
          "search_tag": Array [
            "",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
      ],
      "answers_type": "input",
      "breadcrumb": "Цена",
      "clear_params": Array [],
      "image_id": "price",
      "question": "Вопрос с пользовательским вводом",
      "unit": "₽",
    },
    Object {
      "answers": Array [
        Object {
          "answer": "Да",
          "body_type_group": Array [
            "SEDAN",
            "LIFTBACK",
          ],
          "body_type_group_exclude": Array [
            "HATCHBACK_5_DOORS",
          ],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "big-trunk",
            "",
          ],
          "search_tag_exclude": Array [
            "",
          ],
          "skip_next": "",
        },
        Object {
          "answer": "Нет",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
      ],
      "answers_type": "radio",
      "breadcrumb": "Семейное авто",
      "clear_params": Array [],
      "image_id": "family",
      "question": "Вопрос с исключением",
      "unit": "",
    },
    Object {
      "answers": Array [
        Object {
          "answer": "Да",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "catalog_filter": "mark=AUDI,model=A4",
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
        Object {
          "answer": "Нет",
          "body_type_group": Array [],
          "body_type_group_exclude": Array [
            "",
          ],
          "clear_params": Array [],
          "default_value": "",
          "exclude_catalog_filter": "mark=AUDI,model=A4",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [],
          "search_tag_exclude": Array [
            "",
          ],
          "skip_next": "",
        },
      ],
      "answers_type": "radio",
      "breadcrumb": "Ауди или нет?",
      "clear_params": Array [],
      "image_id": "family",
      "question": "Вопрос с каталог фильтром",
      "unit": "",
    },
    Object {
      "answers": Array [
        Object {
          "answer": "первый вариант",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "wide-back-seats",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "+",
        },
        Object {
          "answer": "второй вариант",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "big-trunk",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
        Object {
          "answer": "третий вариант",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "big-trunk",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
      ],
      "answers_type": "checkbox",
      "breadcrumb": "Большой авто",
      "clear_params": Array [],
      "image_id": "",
      "question": "Вопрос с чекбоксами",
      "unit": "",
    },
    Object {
      "answers": Array [
        Object {
          "answer": "Да",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "catalog_filter": "mark=AUDI,model=A4",
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
        Object {
          "answer": "Нет",
          "body_type_group": Array [],
          "body_type_group_exclude": Array [
            "",
          ],
          "clear_params": Array [],
          "default_value": "",
          "exclude_catalog_filter": "mark=AUDI,model=A4",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [],
          "search_tag_exclude": Array [
            "",
          ],
          "skip_next": "",
        },
      ],
      "answers_type": "radio",
      "breadcrumb": "Ауди или нет еще раз спрашиваю?",
      "clear_params": Array [],
      "image_id": "family",
      "question": "Вопрос с каталог фильтром 2",
      "unit": "",
    },
    Object {
      "answers": Array [
        Object {
          "answer": "первый вариант",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "wide-back-seats",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
        Object {
          "answer": "\\"второй вариант \\"",
          "body_type_group": Array [
            "",
          ],
          "body_type_group_exclude": Array [],
          "clear_params": Array [],
          "default_value": "",
          "ready_prefix": "",
          "ready_text": "",
          "search_param": "",
          "search_tag": Array [
            "wide-back-seats",
          ],
          "search_tag_exclude": Array [],
          "skip_next": "",
        },
      ],
      "answers_type": "radio",
      "breadcrumb": "Широкий задний диван",
      "clear_params": Array [],
      "image_id": "",
      "question": "Вопрос с выбором одного ответа",
      "unit": "",
    },
  ],
}
`;
